From 7a5df704adbc611f15e46c174dc7cc721ccd7cca Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Tue, 28 Nov 2023 15:56:07 -0800
Subject: [PATCH] rbutil: fix compilation on non-glibc Linux systems

Other libc implementations, like musl, may not support this non-POSIX
type.
---
 utils/sansapatcher/sansaio.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/utils/sansapatcher/sansaio.h b/utils/sansapatcher/sansaio.h
index 61e2f1d1b2..5d97211ca9 100644
--- a/utils/sansapatcher/sansaio.h
+++ b/utils/sansapatcher/sansaio.h
@@ -35,7 +35,7 @@
 #define O_BINARY 0
 
 /* Only Linux seems to need lseek64 and loff_t */
-#if !defined(linux) && !defined (__linux)
+#if !defined(linux) && !defined (__linux) || !defined(__GLIBC__)
 #define loff_t off_t
 #define lseek64 lseek
 #endif
-- 
2.43.0

